src/main/java/edu/cornell/kfs/module/purap/document/web/struts/IWantDocumentAction.java

```diff
@@ -911,6 +911,31 @@ private void saveUserOption(String principalId, String userOptionName, String us
         }
     }
 
+    // This logic is based upon the routing-succeeded check from the AccountsPayableActionBase.route() method.
+    private boolean documentRoutingSucceeded() {
+        return KNSGlobalVariables.getMessageList().stream().anyMatch(
+                message -> StringUtils.equals(message.getErrorKey(), KFSKeyConstants.MESSAGE_ROUTE_SUCCESSFUL));
+    }
+
+    /**
+     * Overridden to skip the missing-attachments prompt when submitting an IWNT document
+     * via the 4-step process, because that process already performs an equivalent prompt
+     * after Step 3 if necessary.
+     * 
+     * If we ever update the IWNT document to include additional prompts, then we may need
+     * to revisit this workaround or revisit the Step-3-prompting logic.
+     */
+    @Override
+    public ActionForward promptBeforeValidation(
```

Comment:
I think this approach makes sense here.  I suggest adding a note to the IWantDocumentPreRules.doPrompts() noting if additional prompts are added, this function will need to be thought about too

Reply:
Ok, I'll add a note there accordingly.